src/compiler/crystal/tools/print_hierarchy.cr

```diff
@@ -3,8 +3,15 @@ require "colorize"
 require "../syntax/ast"
 
 module Crystal
-  def self.print_hierarchy(program, exp)
-    HierarchyPrinter.new(program, exp).execute
+  def self.print_hierarchy(program, exp, format)
+    case format
```

Comment:
I moved raising the error to when we are setting the output format, before compiling.